### PR TITLE
Branch v1.4 developerguide architecture

### LINKED ISSUE #235: Update Model and Ui of Architecture Diagram

### LINKED ISSUE #236: Update all DG links to point to this fork instead of upstream repo

### LINKED ISSUE #241: Update Model and Ui Component Description in DG

### PR DESCRIPTION
**Pull request by Droffilc13**
* Update Model and Ui component description
* Update Model architecture diagram
image

fix #235 
fix #236 
fix #241  

Others:
* Edit the README.md page

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -131,30 +131,28 @@ How the parsing works:
...
+**API** : [`Model.java`](https://github.com/AY2122S1-CS2103T-T13-2/tp/blob/master/src/main/java/seedu/address/model/Model.java)
 
 <img src="images/ModelClassDiagram.png" width="450" />
 
 
 The `Model` component,
 
-* stores the address book data i.e., all `Person` objects (which are contained in a `UniquePersonList` object).
+* stores the versioned address book data which includes
+  * all 'Person' objects contained in a `UniquePersonList` object
```
Comment (qinguorui2001) [inline docs/DeveloperGuide.md:143]: `Person` not 'Person'

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -131,30 +131,28 @@ How the parsing works:
...
+  * `Assignment` objects of the "active" `Person` object contained in a `UniqueAssignmentList` object.
```
Comment (qinguorui2001) [inline docs/DeveloperGuide.md:144]: inconsistent use of "." at the end of sentence

Comment (qinguorui2001) [reply docs/DeveloperGuide.md:144]: also, all  `Assignment` objects

File: docs/index.md [hunk 3]
```diff
@@ -10,13 +11,13 @@ title: Introducing to you: TA<sup>2</sup>
...
 
-**Teaching Assistant's Assistant(TA<sup>2</sup>) is a desktop productivity application developed for teaching assistants to make their teaching journey smoother.**
+**Teaching Assistant's Assistant (TA<sup>2</sup>) is a desktop productivity application developed for teaching assistants to make their teaching journey smoother.**
 With TA<sup>2</sup>, you can manage student information and keep track of the status of their assignments. Just type in
 the corresponding commands and TA<sup>2</sup> will manage all these for you! No need to do all these manually!
 
 ## How is TA<sup>2</sup> different?
 
-TA<sup>2</sup> is optimised for use via a Command Line Interface(CLI) while still having the benefits of a Graphical User Interface(GUI). 
+TA<sup>2</sup> is optimised for use via a Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI).
```
Comment (qinguorui2001) [inline docs/index.md:20]: not sure which spacing format is better

Comment (qinguorui2001) [reply docs/index.md:20]: but if use, should be consistent for all develop or user guide

**Review by qinguorui2001 — CHANGES_REQUESTED** [review]
quite good, just few changes to typo

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Droffilc13) [reply docs/DeveloperGuide.md:142]: Good catch

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Droffilc13) [reply docs/DeveloperGuide.md:143]: Resolved all, thanks for pointing out

File: docs/index.md (hunk 3 above)
Comment (Droffilc13) [reply docs/index.md:20]: yep, we should have space, visually more aesthetic

**Review by qinguorui2001 — APPROVED** [review]
LGTM

**Pull request merged**